src/json-patch-duplex.ts

```diff
@@ -400,7 +404,7 @@ module jsonpatch {
     for (var t = oldKeys.length - 1; t >= 0; t--) {
       var key = oldKeys[t];
       var oldVal = mirror[key];
-      if (obj.hasOwnProperty(key)) {
+      if (obj.hasOwnProperty(key) && !(_isArray(obj) === false && obj[key] === undefined)) {
```

Comment:
I haven't because I have nothing to compare it to. We have current behaviour that is wrong but fast, and new behaviour which is good but slower.

One thing for sure, the condition should be reverted to: `(obj[key] === undefined && _isArray(obj) === false)`